ggshield/dev_scan.py

```diff
@@ -25,37 +26,50 @@ def cd(newdir):
         os.chdir(prevdir)
 
 
+def scan_repo_path(
+    client: GGClient, config: Config, repo_path: str
+) -> int:  # pragma: no cover
+    try:
+        with cd(repo_path):
+            return scan_commit_range(
+                client=client,
+                commit_list=get_list_all_commits(),
+                verbose=config.verbose,
+                filter_set=path_filter_set(Path(os.getcwd()), []),
+                matches_ignore=config.matches_ignore,
+                all_policies=config.all_policies,
+                show_secrets=config.show_secrets,
+            )
+    except click.exceptions.Abort:
+        return 0
+    except Exception as error:
+        if config.verbose:
+            traceback.print_exc()
+        raise click.ClickException(str(error))
+
+
 @click.command()
 @click.argument("repository", nargs=1, type=click.STRING, required=True)
 @click.pass_context
 def repo_cmd(ctx: click.Context, repository: str) -> int:  # pragma: no cover
     """
-    clone and scan a REPOSITORY.
+    scan a REPOSITORY at a given URL or path
+
+    REPOSITORY is the clone URI or the path of the repository to scan.
+    Examples:
 
-    REPOSITORY is the clone URI of the repository to scan.
-    example:
     ggshield scan repo git@github.com:GitGuardian/gg-shield.git
+
+    ggshield scan repo /repositories/gg-shield
     """
-    config = ctx.obj["config"]
-    try:
+    config: Config = ctx.obj["config"]
+    client: GGClient = ctx.obj["client"]
+    if os.path.isdir(repository):
+        return scan_repo_path(client, config, repository)
+    else:
```

Comment:
We could check if the argument is a valid URI before trying to clone, to give the user a more explicit error message, like:
`Given argument is neither a valid path nor a valid URL`